src/api/c/sparse.cpp

```diff
@@ -398,7 +398,7 @@ af_err af_sparse_get_values(af_array *out, const af_array in)
 af_err af_sparse_get_row_idx(af_array *out, const af_array in)
 {
     try {
-        const SparseArrayBase base = getSparseArrayBase(in);
+        const SparseArrayBase &base = getSparseArrayBase(in);
```

Comment:
I am not sure this is a valid change. I think the idea is that you do want to make a copy.

This is also unrelated to the name of the PR. If you want to do this you should create a new PR.

Reply:
Shouldn't we be returning simple object in that case instead of reference to an object ? The reason I say that is `getSparseArrayBase` currently returns an object of type `SparseArrayBase&`. It is similar case with `getInfo` too - returns `const ArrayInfo`.